apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -1402,8 +1402,17 @@ private void HandleAckResponse(Task writeTask, List<string> ackIds, List<string>
                     }
                     else
                     {
-                        // Unrecoverable error; throw exception.
-                        throw writeTask.Exception.FlattenIfPossible();
+                        // Check if its RpcException with StatusCode of InvalidArguments and Reason as EXACTLY_ONCE_ACKID_FAILURE.
+                        // If it is, then ignore it and continue as they are fire and forget. We may want to log them later. 
+                        // Other non-recoverable errors will continue to be thrown.
+                        var rpcException = writeTask.Exception.As<RpcException>();
+                        bool ignoreError = rpcException?.StatusCode == StatusCode.InvalidArgument &&
```

Comment:
So the ack/modack behavior should be:
1) Retry Unavailable errors for ack/modack (this already happens at the GAPIC/GRPC level, see [here](https://github.com/googleapis/googleapis/blob/480e3fbafbe4eb61c2118e3cb8d9e0d8eb9a9e36/google/pubsub/v1/pubsub_grpc_service_config.json#L221))
2) Ignore all GRPC errors for ack/modack

Please change the code to ignore all GRPC errors for ack/modack in this PR.

The reason we ignore GRPC errors for ack/modack is because they are best-effort RPCs. The user shouldn't see an exception when these happen. There's a subtlety - what if the subscription is not accessible (PermissionDenied) or doesn't exist (NotFound)? Our thinking around this is that it's not that important that the user sees these errors with ack/modack because the StreamingPull stream will fail if these errors occur and then the whole subscriber object will fail. So no need to do extra work here. We may revisit this choice for these two special error codes though. For now, we'll stick to this in the name of uniformity (the Java client has the behavior described here.)

Once we do the full exactly-once delivery work, the whole behavior will be:
1) Retry Unavailable errors for ack/modack at the GAPIC/GRPC level (no change to this behavior)
2) If exactly-once is disabled, ignore all GRPC errors for ack/modack
3) If exactly-once is enabled, bubble up extra info via futures.

Reply:
Swallowed all the gRPC errors in Ack/ModAck call. Have tested thoroughly with a client app for an hour. Created new issue #8291 for writing unit tests as it may require substantial time.

Reply:
Thank you Rishabh, especially for the thorough testing. Please write the unit tests in this PR itself. We usually prefer it that way unless there's a strong reason otherwise. Yes, this issue is high priority, but we can wait for unit tests.

Reply:
Unit Tests are now added. Please have a look.